lib/suse/connect/zypper.rb

```diff
@@ -44,9 +45,47 @@ def add_service(service_url, service_name)
           call("--non-interactive addservice -t ris #{service}")
         end
 
-        # @param service_name [String] Alias-mnemonic with which zypper should add this service
+        # @param service_name [String] Alias-mnemonic with which zypper should remove this service
         def remove_service(service_name)
           call("--non-interactive removeservice '#{Shellwords.escape(service_name)}'")
+          remove_service_credentials(service_name)
+        end
+
+        ##
+        # Remove all installed SUSE services
+        def remove_all_suse_services
+          services.each do |service|
+            if service[:url].include?(Config.new.url)
+              remove_service(service[:name])
+            end
+          end
+        end
+
+        # @param service_name [String] Alias-mnemonic with which service credentials file should be removed
+        def remove_service_credentials(service_name)
+          service_credentials_file = File.join(SUSE::Connect::Credentials::DEFAULT_CREDENTIALS_DIR, service_name)
+
+          if File.exist?(service_credentials_file)
+            File.delete service_credentials_file
+          end
+        end
+
+        ##
+        # Returns an array of hashes of all installed services
+        def services
+          output = call('services -u', false)
```

Comment:
i think it would be better to parse the xmloutput here (`zypper -x services`). 
So we are more robust if zypper decides to change its text output format.